executor/prepared.go

```diff
@@ -336,15 +336,23 @@ func ResetStmtCtx(ctx context.Context, s ast.StmtNode) {
 	sc.TimeZone = sessVars.GetTimeZone()
 
 	switch stmt := s.(type) {
-	case *ast.UpdateStmt, *ast.DeleteStmt:
+	case *ast.UpdateStmt:
 		sc.IgnoreTruncate = false
 		sc.OverflowAsWarning = false
 		sc.TruncateAsWarning = !sessVars.StrictSQLMode
 		sc.InUpdateOrDeleteStmt = true
+		sc.IgnoreError = stmt.IgnoreErr
+	case *ast.DeleteStmt:
+		sc.IgnoreTruncate = false
+		sc.OverflowAsWarning = false
+		sc.TruncateAsWarning = !sessVars.StrictSQLMode
+		sc.InUpdateOrDeleteStmt = true
+		sc.IgnoreError = stmt.IgnoreErr
```

Comment:
1. According to https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#ignore-strict-comparison
we may **not** need an IgnoreErr flag in StatementContext,
we can check node.IgnoreErr here, and set the IgnoreTruncate/ TruncateAsWarning to the certain value,
2. We may need **IgnoreDividedByZero and DividedByZero** flags in StatementContext since TiDB does not support a full implementation of SQL_MODE now.

Reply:
1. Why we need *DividedByZero* flag in StatementContext ? For the following process?
2. I think DividedByZeroAsWarning instead of IgnoreDividedByZero is better name. Because we have IgnoreTruncate and TruncateAsWarning, and they have different meanings.